File: packages/emitter/testcases/types/dictionaries/Dictionaries.cs
namespace TestCases.types.dictionaries
{
        public static class Dictionaries
        {
            public static global::System.Collections.Generic.Dictionary<string, double> getStringDict()
                {
                return new global::System.Collections.Generic.Dictionary<string, double>();
                }

            public static global::System.Collections.Generic.Dictionary<double, string> getNumberDict()
                {
                return new global::System.Collections.Generic.Dictionary<double, string>();
                }

            // type NumberIndexed = global::System.Collections.Generic.Dictionary<double, string>

            public static string? lookupByNumber(global::System.Collections.Generic.Dictionary<double, string> dict, double key)
                {
                return dict[key];
                }
        }
}

File: packages/targets/csharp/emitter/testcases/common/types/dictionaries/Dictionaries.ts
// Dictionary with string keys (existing behavior)
export function getStringDict(): Record<string, number> {
  return {};
}

// Dictionary with number keys (new)
export function getNumberDict(): Record<number, string> {
  return {};
}

// Index signature with number key
export interface NumberIndexed {
  [key: number]: string;
}

// Function using number-keyed dictionary
export function lookupByNumber(
  dict: Record<number, string>,
  key: number
): string | undefined {
  return dict[key];
}
